roles/argocd/templates/argocd-configmap.yaml.j2

```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: argocd-cm
```

Comment:
Where does argocd read and use these credentials? I couldn't find a reference to this configmap anywhere (or am I misunderstanding something?)

Or does argocd know to search its namespace for the `repositories` key?

Reply:
Ah, this might be legacy behaviour? https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/#legacy-behaviour

From reading the docs it could be simpler to just have the secrets but specify the `url` there? 

https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/#repositories

e.g. 
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: seldon-gitea-secret
  namespace: {{ argocd_namespace }}
  labels:
    argocd.argoproj.io/secret-type: repository
stringData:
  url: {{ gitops_repository }} 
  password: {{ gitops_password }}
  username: {{ gitops_username }}
```

Reply:
This here is required to reference the `seldon-gitea-repo`. 
This configmap explicitly refrences the secret 
```
      passwordSecret:
        name: seldon-gitea-repo
        key: password
      usernameSecret:
        name: seldon-gitea-repo
        key: username
```

The link you provided leads to "latest" version of ArgoCD (2.0.5 I think) and for some reason this one didn't want to work at all. 

The way you copy-pasted it here seems like a new way of configuring it that didn't work for me with 2.0.4. 

Having secret and configmap entry works properly

Reply:
Kind of explained here https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#https-repositories